hydroflow_lang/src/graph/hydroflow_graph.rs

```diff
@@ -411,16 +411,14 @@ impl HydroflowGraph {
     /// in the above eaxmple, the \[myport\] port will be used to connect the source_iter with the map that is inside of the module.
     /// The output module boundary has elided ports, this is also used to match up the input/output across the module boundary.
     pub fn merge_modules(&mut self) -> Result<(), Diagnostic> {
-        let mut to_remove = Vec::new();
-
-        for (nid, node) in self.nodes() {
-            if matches!(node, GraphNode::ModuleBoundary { .. }) {
-                to_remove.push(nid);
-            }
-        }
-
-        for nid in to_remove {
-            self.remove_module_boundary(nid)?;
+        let mod_bound_nodes = self
```

Comment:
Do you need to collect the result into a vec? can mod_bound_nodes be an iterator?

Reply:
Unfortunately since it keeps a reference to `self` alive, the iterator can't be alive at the same time as the call to `remove_module_boundary` (which needs an exclusive ref)